src/backend/distributed/commands/database.c

```diff
@@ -192,6 +192,22 @@ PreprocessGrantOnDatabaseStmt(Node *node, const char *queryString,
 }
 
 
+static bool
+isSetTablespaceStatement(AlterDatabaseStmt *stmt)
+{
+	ListCell *lc = NULL;
+	foreach(lc, stmt->options)
+	{
+		DefElem *def = (DefElem *) lfirst(lc);
```

Comment:
We use foreach_ptr / foreach_oid etc. instead of plain foreach in %95 of the codebase, we use foreach only if other choices don't fit.

```suggestion
	DefElem *def = NULL;
	foreach_ptr(def, stmt->options)
	{
```

Reply:
Fixed

Reply:
Fixed